src/test/java/spoon/generating/CloneVisitorGenerator.java

```diff
@@ -251,11 +251,40 @@ private CtInvocation createFactoryInvocation(CtVariableAccess<CtElement> element
 					if (isSubTypeOfCtElement(ctField.getType())) {
 						continue;
 					}
-					final CtMethod<?> setterOfField = getSetterOf(ctField);
-					final CtInvocation<?> setterInvocation = createSetterInvocation(//
-							element.getParameters().get(0).getType(), setterOfField, //
-							createGetterInvocation(element.getParameters().get(0), getGetterOf(ctField)));
-					final List<CtMethod<?>> methodsToAvoid = getCtMethodThrowUnsupportedOperation(setterOfField);
+					CtMethod<?> setterOfField = getSetterOf(ctField);
+
+					CtMethod<?> getter = getGetterOf(ctField);
+					// The invocation that fetches the value to set in the clone. This can be a simple getter
+					// call or it can modify the result (e.g. to clone it) after retrieving it.
+					CtInvocation<?> fetchValueInvocation = createGetterInvocation(
+							element.getParameters().get(0),
+							getter
+					);
+
+					// CtExtendedModifiers are treated specially throughout this class. When cloning an object
+					// we need to clone the extended modifiers as well. This is not handled by the
+					// CloneVisitor, as CtExtendedModifiers are no CtElement and therefore our CloneBuilder
+					// needs to do that itself.
+					// This rewrites the getter invocation:
+					//   `original.getExtendedModifiers()` to `clone(original.getExtendedModifiers())`
+					// The clone method is defined in the CloneBuilderTemplate.
+					if (getter.getSimpleName().equals("getExtendedModifiers")) {
+						CtExecutableReference<Object> cloneReference = factory.Executable()
+								.createReference(
+										"Set<CtExtendedModifier> #clone(Set<CtExtendedModifier>)"
+								);
+						fetchValueInvocation = factory.Code().createInvocation(
+								null, cloneReference, fetchValueInvocation
+						);
+					}
+
+					CtInvocation<?> setterInvocation = createSetterInvocation(
+							element.getParameters().get(0).getType(),
+							setterOfField,
+							fetchValueInvocation
+					);
+
+					List<CtMethod<?>> methodsToAvoid = getCtMethodThrowUnsupportedOperation(setterOfField);
```

Comment:
I suggest something like this:

```suggestion
					CtParameter<?> receiver = element.getParameters().get(0);
					CtMethod<?> setter = getSetterOf(ctField);
					CtMethod<?> getter = getGetterOf(ctField);
					CtInvocation<?> setterInvocation = createSetterInvocationWithGetterReturnValueAsArgument(
							receiver, getter, setter);
							
					List<CtMethod<?>> methodsToAvoid = getCtMethodThrowUnsupportedOperation(setter);
```

With the methods defined like so:

```java
			private CtInvocation<?> createFetchValueInvocation(CtParameter<?> receiver, CtMethod<?> getter) {
			    CtInvocation<?> getterInvocation = createGetterInvocation(receiver, getter);

				if (getter.getSimpleName().equals("getExtendedModifiers")) {
					// TODO explain here why we wrap the getter in a clone
					return wrapExtendedModifiersGetterInvocationInClone(getterInvocation);
				} else {
					return getterInvocation;
				}
			}

			private CtInvocation<?> wrapExtendedModifiersGetterInvocationInClone(CtInvocation<?> getterInvocation) {
				CtExecutableReference<Object> cloneReference = factory.Executable()
						.createReference(
								"Set<CtExtendedModifier> #clone(Set<CtExtendedModifier>)"
						);
				return factory.Code().createInvocation(null, cloneReference, getterInvocation);
			}
```

It's essentially what you initially wrote, but with the branching on `getExtendedModifiers` pulled up just one level above `createGetterInvocation. WDYT?

Reply:
I don't love those names but as I said earlier I didn't come up with anything pretty either. I used your solution and added some (probably too verbose) comments which are hopefully helpful enough.

Reply:
The original factoring of the method is, let's be honest, not great. To better isolate the behaviors and put more descriptive names on them, I'd say the whole thing would need to be refactored.

Reply:
And I agree with the verbosity of the comments being a tad bit high. But, this is not a part of the code base that is touched often, and so the risk of them going out of date isn't that large.

As a note, I personally prefer putting a brief comment and then adding a link to the related issue for more context.